src/bin/pgcopydb/pgsql_timeline.c

```diff
@@ -0,0 +1,363 @@
+/*
+ * src/bin/pgcopydb/pgsql_timeline.c
+ *	 API for sending SQL commands about timelines to a PostgreSQL server
+ */
+
+#include "file_utils.h"
+#include "log.h"
+
+#include "pgsql.h"
+#include "pgsql_utils.h"
+#include "catalog.h"
+
+static void parseIdentifySystemResult(void *ctx, PGresult *result);
+static void parseTimelineHistoryResult(void *ctx, PGresult *result);
+
+
+typedef struct IdentifySystemResult
+{
+	char sqlstate[6];
+	bool parsedOk;
+	IdentifySystem *system;
+} IdentifySystemResult;
+
+
+typedef struct TimelineHistoryResult
+{
+	char sqlstate[6];
+	bool parsedOk;
+	char filename[MAXPGPATH];
+	char content[BUFSIZE * BUFSIZE]; /* 1MB should get us quite very far */
```

Comment:
That should now be a dynamic memory allocation, otherwise I don't think this PR will solve the bug it's been assigned to fix?

Reply:
Fair point. This piece of memory needs to be handled dynamically.

The reported bug was about the limitation on the number of lines (PGCOPYDB_MAX_TIMELINES). This line is about the total size of the timeline file (PGCOPYDB_MAX_TIMELINE_CONTENT).

On that note, I will remove both of these constants in a following commit as they are not used anywhere in the code.